### ISSUE TITLE
Not working with AMS 2.

### ISSUE THREAD

**Issue by DKHSG** [fholger/vrperfkit#16]
Copied the files to the .exe directory, but it`s not working. There isn`t even a logfile after starting the game.

**Comment by fholger** [comment]
If there is no log file, the mod wasn't loaded. The most likely cause is that you haven't put the files in the right directory. Search the directory structure if there is another executable somewhere deeper in the path, e.g. something like `Bin\win64` or `Engine` or anything of the kind.

**Comment by DKHSG** [comment]
I`m sure it`s in the same directory as the AMS2.exe file. Reshade is working correct with the .dll in the same directory. I removed all reshade files to test vrperfkit.

**Comment by fholger** [comment]
I'm afraid I can't offer any further assistance. I don't own the game. But if there is no log file, that means that the game did not load vrperfkit's DLL. And I can think of no explanation that would make the game load Reshade's dxgi.dll, but not vrperfkit's.

**Comment by DKHSG** [comment]
For AMS2 i had to rename the .ddl from reshade(vrtoolkit from retrolux) to dinput8.dll. When i rename your .ddl the game is not starting.

**Comment by fholger** [comment]
Ah, yes. vrperfkit only works as either dxgi.dll or d3d11.dll, nothing else. If the game does not pick it up that way, then it's not using D3D11 and would not work with vrperfkit, anyway.

**Comment by nabel0** [comment]
In on Oculus and vrperfkit is working fine for me for AMS2.
Are you running it over SteamVR (openvr) or over the Oculus API?

**Comment by DKHSG** [comment]
SteamVR and WMR.

**Comment by nabel0** [comment]
Thats strange...I can only say it works via the oculus api:

21:32:32 [17760] ======================
21:32:32 [17760] VR Performance Toolkit
21:32:32 [17760] ======================

21:32:32 [17760] Current configuration:
21:32:32 [17760]   Upscaling (FSR) is enabled
21:32:32 [17760]     * Render scale: 0.77
21:32:32 [17760]     * Sharpness:    0.9
21:32:32 [17760]     * Radius:       0.5
21:32:32 [17760]     * MIP bias:     enabled
21:32:32 [17760]   Debug mode is disabled
21:32:32 [17760] Currently active hotkeys:
21:32:32 [17760]   * cycleUpscalingMethod 17 113
21:32:32 [17760]   * increaseUpscalingRadius 17 114
21:32:32 [17760]   * decreaseUpscalingRadius 17 115
21:32:32 [17760]   * increaseUpscalingSharpness 17 116
21:32:32 [17760]   * decreaseUpscalingSharpness 17 117
21:32:32 [17760]   * toggleDebugMode 17 112
21:32:32 [17760]   * toggleUpscalingApplyMipBias 17 118
21:32:32 [17760]   * toggleFixedFoveated 18 112
21:32:32 [17760]   * captureOutput 17 119
21:32:32 [17760] Installing hook for LoadLibraryA from 00007FFAEA71F500 to 00007FFAC7B7B2C0
21:32:32 [17760] Installing hook for LoadLibraryExA from 00007FFAEA71D680 to 00007FFAC7B7B310
21:32:32 [17760] Installing hook for LoadLibraryW from 00007FFAEA71E880 to 00007FFAC7B7B3F0
21:32:33 [17760] Installing hook for LoadLibraryExW from 00007FFAEA7193F0 to 00007FFAC7B7B380
21:32:33 [17760] LibOVRRT64_1.dll is loaded in the process, installing hooks...
21:32:33 [17760] Installing hook for ovr_Initialize from 00007FFA81BC8530 to 00007FFAC7B5F090
21:32:33 [17760] Installing hook for ovr_GetFovTextureSize from 00007FFA81BC95F0 to 00007FFAC7B5EFA0
21:32:33 [17760] Installing hook for ovr_EndFrame from 00007FFA81BC9980 to 00007FFAC7B5EE20
21:32:33 [17760] Installing hook for ovr_SubmitFrame from 00007FFA81BC9A70 to 00007FFAC7B5F300
21:32:33 [17760] Installing hook for ovr_SubmitFrame2 from 00007FFA81BC9BC0 to 00007FFAC7B5F180
21:32:33 [17760] d3d11.dll is loaded in the process, installing hooks...
21:32:33 [17760] Installing hook for D3D11CreateDevice from 00007FFAE0DD5440 to 00007FFAC7B5CBA0
21:32:33 [17760] Installing hook for D3D11CreateDeviceAndSwapChain from 00007FFAE0DD55C0 to 00007FFAC7B5D020
21:32:33 [17760] Oculus runtime initialization for version 3
21:32:33 [17760] Loading DLL at "C:\\WINDOWS\\system32\\dxgi.dll"
21:33:21 [17760] Initializing Oculus frame submission...
21:33:21 [17760] Game is using D3D11 swapchains, initializing D3D11 resources
21:33:21 [17760] Swap chain has format 5, bind flags 1 and misc flags 8
21:33:21 [17760] Eye 0: submitted textures have resolution 1824x1856
21:33:21 [17760] Eye 0: output resolution is 2370x2410
21:33:21 [17760] Creating output swapchain in format 4
21:33:21 [17760] Swap chain has format 5, bind flags 1 and misc flags 8
21:33:21 [17760] Eye 1: submitted textures have resolution 1824x1856
21:33:21 [17760] Eye 1: output resolution is 2370x2410
21:33:21 [17760] Creating output swapchain in format 4
21:33:21 [17760] Trying to load NVAPI...
21:33:21 [17760] Successfully initialized NVAPI; Variable Rate Shading is available.
21:33:21 [17760] Installing virtual function hook for ID3D11DeviceContext::PSSetSamplers
21:33:21 [17760] Installing virtual function hook for ID3D11DeviceContext::OMSetRenderTargets
21:33:21 [17760] Installing virtual function hook for ID3D11DeviceContext::OMSetRenderTargetsAndUnorderedAccessViews
21:33:21 [17760] D3D11 resource creation complete
21:33:21 [17760] Creating D3D11 resources for FSR upscaling...
21:33:21 [17760] Creating replacement sampler for 00000000104C5EA0 with MIP LOD bias -0.38
21:33:21 [17760] Creating replacement sampler for 000000004B616DA0 with MIP LOD bias -0.38
21:33:21 [17760] Creating replacement sampler for 00000000104C66A0 with MIP LOD bias -0.38
21:33:21 [17760] Creating replacement sampler for 0000000007C83B20 with MIP LOD bias -0.38
21:33:21 [17760] Creating replacement sampler for 00000000104C69E0 with MIP LOD bias -0.38
21:33:21 [17760] Creating replacement sampler for 000000004B616CA0 with MIP LOD bias -0.38
21:33:21 [17760] Creating replacement sampler for 0000000007C83220 with MIP LOD bias -0.38
21:33:21 [17760] Creating replacement sampler for 00000000104C6220 with MIP LOD bias -0.38
21:33:21 [17760] Creating replacement sampler for 00000000104C5FE0 with MIP LOD bias -0.38
21:33:21 [17760] Creating replacement sampler for 0000000007C83C60 with MIP LOD bias -0.38
21:33:21 [17760] Creating replacement sampler for 00000000104C6460 with MIP LOD bias -0.38

Why it wont work via the openvr api is strange to me.

**Comment by fholger** [comment]
What happens if you *don't* rename the vrperfkit DLL and leave it at dxgi.dll? Did you try that? Does it still not produce a log file?

**Comment by DKHSG** [comment]
I tried that first. No log file and no effect.

**Comment by fholger** [comment]
Well, if that approach is working for nabel0, then one of you is lying or doing something wrong :D It either loads or doesn't. It can't work on one machine and not on the other.

**Comment by nabel0** [comment]
> Well, if that approach is working for nabel0, then one of you is lying or doing something wrong :D It either loads or doesn't. It can't work on one machine and not on the other.

The difference here is that he is using steamvr api and i am using oculus api...Maybe thats the point.

**Comment by fholger** [comment]
No, that has nothing to do with whether the DLL is loaded or not. Loading the DLL happens way before the game ever initializes either Oculus or SteamVR.

**Comment by nabel0** [comment]
hmm...i think this game does something special...Lots of ppl reporting that it dont work if they use steamVR...
Strange...

--> https://forum.reizastudios.com/threads/fsr-for-openvr.19036/page-5

**Comment by robertjamesdavies** [comment]
I'm still playing now. A colleague with AMS2/G2/SteamVR has it working, but I don't.... just don't fire... so will report back if I manage to work out why.

**Comment by JCae2798** [comment]
Steam user here as well reporting in the forums this is not working for AMS2.  If you check the original post, they created a custom version from your original mod as it never worked to begin with.  We as users have no idea what the changes were done to the custom file but whatever it was it worked that way only.  Is there anyone with knowledge of what the original person from the forum did?  Unfortunately i don't have experience with modding these files.


a copy of the modded file is here:  https://drive.google.com/file/d/1ckOBXtoqWpd5BvSmVS5ttE5Z4PYX2sij/view

**Comment by JCae2798** [comment]
I will add that the openVR dlls sit in the x64 folder structure but the exe is in the game root.  I tried both folders without success.

here's a link to the folder files:  https://www.dropbox.com/s/rzhy7ods6ydm7qw/folder_ams.png?dl=0

**Comment by nabel0** [comment]
You are mixing things up.
Vrperfkit works completely different then the old openvr_fsr mod.
The Vrperfkit files belogs to your .exe of the game. And your previously changed "openvr_api.dll" files needs to be the original one.

**Comment by JCae2798** [comment]
Understood, i was sharing this with fholger so he knows how this games folder structure works.  What i am saying is i have the files in root, but attempted to move them to the x64 folder too to test and neither options works.  But i understand the design is to keep it in root.

I also have the original API dll file loaded.  The custom one i renamed as you can see in my screen shot as a backup.

**Comment by robertjamesdavies** [comment]
Is there a need to have a full version of the NVIDIA driver installed? For any additional API's? I usually don't install Geforce Experience et. al but will give it a go and see.

**Comment by fholger** [comment]
No, Geforce Experience does not make any difference. Not for the mod, anyway, and it shouldn't for the game, either.

**Comment by robertjamesdavies** [comment]
Thanks for quick response - assuming CPU (AMD or Intel) makes no odds (just trying to nail down any differences between my and my friends setup)

Is there a way I could try and diagnose why it’s not being loaded with the game?

**Comment by JCae2798** [comment]
[vrperfkit.log](https://github.com/fholger/vrperfkit/files/7910146/vrperfkit.log)

Great news!  Someone on the forums provided the answer to why it's not loading correctly.  See here:  https://forum.reizastudios.com/threads/fsr-for-openvr.19036/page-5#post-169752

So now i see the log file, but it still seems i cannot see the effects or use the hot keys.  LOG file attached above.

My guess is the problem is as follows?:
from log:
.....
22:20:54 [8784] IVRClientCore::Cleanup was called, deleting hooks...
22:20:54 [8784] Removing hook to 00007FF81FE8E180
22:20:54 [8784] Removing hook to 00007FF81FE8E0A0
22:20:54 [8784] Removing hook to 00007FF81FEB1AD0
22:20:54 [8784] Removing hook to 00007FF81FEB7F50

**Comment by fholger** [comment]
According to the log, it never submitted any images to the headset. Just to be clear, vrperfkit does *not* work on the desktop mirror, it is only activated once headset rendering starts. So if AMS2 is like e.g. Assetto Corsa, you need to get into an actual race before anything happens.

If you did that and that log is from such a run, then yeah, vrperfkit never saw Submit being called. Removing the hooks is by design and should not cause that as they are reinstalled as necessary, but I guess if the game is doing something weird, it could miss some? You can try this build, but I wouldn't be super hopeful
[dxgi.zip](https://github.com/fholger/vrperfkit/files/7912225/dxgi.zip)
.

**Comment by robertjamesdavies** [comment]
Hey @fholger - many thanks for your continued help here. I tried to build above, but no difference as yo said.

However, following the instructions here has resulted in the mod being loaded.... no idea why! - https://forum.reizastudios.com/threads/fsr-for-openvr.19036/page-5#post-169752 

Basically using the ASI side-loader here - https://github.com/ThirteenAG/Ultimate-ASI-Loader

**Comment by JCae2798** [comment]
OK i'm a dummy.  Testing again today I'm seeing the improvements to FPS from the mod and I am getting it to load using the above method by using AIS Loader, but I am seeing the following compared to the previous mod.  Wondering if its by design or something else?  Maybe the AIS Loader isn't allowing the mod to work correctly...

1. Using same config settings from old mod to new one, the older mod shows sharper image which shows in FPS as well (FPS hit).  No matter though if i change the settings on the mod, i cant seem to get a different image...Even changing from FSR to NIS i see no difference.  Its like it uses a default setting no matter what i list in config.  I do see the correct settings in the LOG file though every time i reload the game, just no changes in the actual image quality from the HMD
2. The hotkeys aren't working no matter what i try.  Could this be a result of the AIS Loader?
3. The new file you shared did work too, but for whatever reason was breaking some of the menus or pop up alerts in game.  Reverting back to the original release DLL resolved this.  I did some some big changes in the LOG file too so if you want to see compare if its helpful i can share.
4. Lastly i don't see any effects of the fixedFoveated.  Whether i enable/disable, or change radius: from 0.1 to 0.9 i see now image differences in the HMD which leads me back to bullet #1.  Again LOG file does report the correct config changes.

Happy to do any further testing and provide any requested data.  I do appreciate your support and also releasing these great mods to the community!!!!!  This has seriously made my Simracing VR experiences so much better.

**Comment by fholger** [comment]
Yes, please provide logs. It sounds like it isn't actually doing upscaling, potentially just lowering the resolution, which would explain the FPS gain without sharpening. The difference between FSR and NIS is very obvious, especially at higher sharpness settings, so if you can't see that, in all likelihood something isn't working right.

**Comment by robertjamesdavies** [comment]
Hey guys.

A few observations from my testing earlier.

- I do believe it is upscaling (but my eyes could be deceiving me as I run at 0.8, so 80% qualtiy) with steam SS at 100% (Reverb G2) - sometimes hard to tell!
- Life you, hotkeys don't work for me.
- Debug mode when set to true does not show the circle in the HMD where the full quality upscaling is applied. Your previous FSR mod did do this (the one NuScorpi re-built) hence me knowing that 0.3 is fine on the G2, which has a poor sweet spot. .
- I was able to notice the difference when changing the sharpening value in the config file. I used NIS (not FSR) and dialled it down from 0.7 to 0.2, which I assume is 20%, in line with a setting I use in a similar NIS upscaling mod for FS2020 - anything above this looks too artifacty! I also dialled back the DX11settings.xml sharpening from 1.8 to 1.2 too.
- So my final settings for my G2 in the Vrperf.cfg are renderscale 0.8, sharpening 0.2, radius 0.3 and foveated rendering on.

Attached is the log file and settings for ref.

Original NuScorpi recompile of your mod here - https://drive.google.com/file/d/1Fb51oHM4g513k5Vs_bO-px4sZMFzA_5p/view?usp=sharing with his comments here https://forum.reizastudios.com/threads/fsr-for-openvr.19036/

**Comment by fholger** [comment]
I'm afraid the log is pretty clear - there's no upscale happening at all. The mod is never seeing a submit call from SteamVR and hence isn't doing any post-processing. That's also why you are not seeing the debug circle and why the hotkeys aren't working.

I don't know why it isn't seeing the submit, but at the moment the mod isn't doing anything. At best, it might be reducing render resolution, I can't be sure of that.

For reference, this line *must* be present in the log to indicate that the mod is actually working:

`21:53:19 [10172] Game is submitting D3D11 textures, creating necessary output resources...`

**Comment by robertjamesdavies** [comment]
Doh, well that’s somewhat embarrassing!

… what next? How can we help you?

**Comment by akhamar** [comment]
dxgi.dll is only loaded when launched in 2D mod.
When launching the game in native VR, dxgi.dll is not loaded at all, most likely because it's a different pipeline used.

**Comment by JCae2798** [comment]
> dxgi.dll is only loaded when launched in 2D mod. When launching the game in native VR, dxgi.dll is not loaded at all, most likely because it's a different pipeline used.
> 
> Steam VR + index

Not sure if it helps, but there are two game EXEs in the root directory where VR uses one and non-VR uses the other.

AMS2.exe vs AMS2AVX.exe

**Comment by robertjamesdavies** [comment]
> I was a "not investor" of Project Cars and PCars2 (access to dev forums), and something in my brain is ringing thinking of VR and the initial Project Cars. I think they did something special to increase performance and reduce rendering issues. Maybe they are still using the old openvr api because of that. But that is me remembering something i briefly read 10 Years ago.

You're right about the old API - NuScorpi responded to my comment on the Reiza forums about it - unfortunately he doesn't have the time to look at adapting this mod. Here's his comment:

"AMS2 / PCARS2 uses an ancient version of OpenVR from just before a change in the API that broke backwards compatibility. That's why adding support originally required a separate version of the mod and wasn't merged into the main code.

I don't have the time to work on this currently but someone did contact me today about the code I wrote so they might be able to produce something that works for the new mod..."

**Comment by mickthompson** [comment]
> I'm afraid the log is pretty clear - there's no upscale happening at all. The mod is never seeing a submit call from SteamVR and hence isn't doing any post-processing. That's also why you are not seeing the debug circle and why the hotkeys aren't working. Whatever difference you think you are seeing between different sharpening values is a placebo.
> 
> I don't know why it isn't seeing the submit, but at the moment the mod isn't doing anything. At best, it might be reducing render resolution, I can't be sure of that.
> 
> For reference, this line _must_ be present in the log to indicate that the mod is actually working:
> 
> `21:53:19 [10172] Game is submitting D3D11 textures, creating necessary output resources...`

I can confirm it is just lowering the resolution, at least according to the fpsVR 'Detect resolution' function.

**Comment by xexets** [comment]
I think I managed to load it with asiloader (both in ASM2 and PC2) and I think I clearly see an improvement in performance in both. Could someone kindly check my log and see if has indeed worked? Thanks!

`12:11:09 [9692] ======================
12:11:09 [9692] VR Performance Toolkit
12:11:09 [9692] ======================

12:11:09 [9692] Current configuration:
12:11:09 [9692]   Upscaling (NIS) is enabled
12:11:09 [9692]     * Render scale: 0.77
12:11:09 [9692]     * Sharpness:    0.7
12:11:09 [9692]     * Radius:       0.4
12:11:09 [9692]     * MIP bias:     enabled
12:11:09 [9692]   Debug mode is disabled
12:11:09 [9692] Currently active hotkeys:
12:11:09 [9692]   * cycleUpscalingMethod 17 113
12:11:09 [9692]   * increaseUpscalingRadius 17 114
12:11:09 [9692]   * decreaseUpscalingRadius 17 115
12:11:09 [9692]   * increaseUpscalingSharpness 17 116
12:11:09 [9692]   * decreaseUpscalingSharpness 17 117
12:11:09 [9692]   * toggleDebugMode 17 112
12:11:09 [9692]   * toggleUpscalingApplyMipBias 17 118
12:11:09 [9692]   * toggleFixedFoveated 18 112
12:11:09 [9692]   * captureOutput 17 119
12:11:09 [9692] Installing hook for LoadLibraryA from 00007FF808A0F500 to 00007FFF6F24B2C0
12:11:09 [9692] Installing hook for LoadLibraryExA from 00007FF808A0D680 to 00007FFF6F24B310
12:11:09 [9692] Installing hook for LoadLibraryW from 00007FF808A0E880 to 00007FFF6F24B3F0
12:11:09 [9692] Installing hook for LoadLibraryExW from 00007FF808A093F0 to 00007FFF6F24B380
12:11:10 [9692] vrclient_x64.dll is loaded in the process, installing hooks...
12:11:10 [9692] Installing hook for VRClientCoreFactory from 00007FFF6BAD1820 to 00007FFF6F2348B0
12:11:10 [9692] d3d11.dll is loaded in the process, installing hooks...
12:11:10 [9692] Installing hook for D3D11CreateDevice from 00007FF800F85690 to 00007FFF6F22CBA0
12:11:10 [9692] Installing hook for D3D11CreateDeviceAndSwapChain from 00007FF800F85810 to 00007FFF6F22D020
12:11:10 [9692] dxgi.dll is loaded in the process, installing hooks...
12:11:10 [9692] Installing hook for CreateDXGIFactory from 00007FF805262FC0 to 00007FFF6F22A9D0
12:11:10 [9692] Installing hook for CreateDXGIFactory1 from 00007FF8052631B0 to 00007FFF6F22AD50
12:11:11 [9692] Installing hook for CreateDXGIFactory2 from 00007FF8052630A0 to 00007FFF6F22B0D0
12:11:11 [9692] Installing hook for DXGIGetDebugInterface1 from 00007FF8052A8590 to 00007FFF6F22C1F0
12:11:11 [9692] Installing hook for DXGIDeclareAdapterRemovalSupport from 00007FF80525E200 to 00007FFF6F22BC30
12:11:11 [9692] OpenVR: requested interface IVRClientCore_002
12:11:11 [9692] Installing virtual function hook for IVRClientCore::GetGenericInterface
12:11:11 [9692] Installing virtual function hook for IVRClientCore::Cleanup
12:11:11 [9692] OpenVR: requested interface IVRSystem_009
12:11:11 [9692] Installing virtual function hook for IVRSystem::GetRecommendedRenderTargetSize
12:11:11 [9692] OpenVR: requested interface IVRCompositor_009
12:11:11 [9692] Installing virtual function hook for IVRCompositor::Submit
12:11:11 [9692] OpenVR: requested interface IVROverlay_007
12:11:11 [9692] IVRClientCore::Cleanup was called, deleting hooks...
12:11:11 [9692] Removing hook to 00007FFF6BA6E3E0
12:11:11 [9692] Removing hook to 00007FFF6BA6E300
12:11:11 [9692] Removing hook to 00007FFF6BA91D30
12:11:11 [9692] Removing hook to 00007FFF6BA981B0
12:17:02 [9692] Shutting down

`

**Comment by fholger** [comment]
Again, the log is showing no traces of any upscaling or FFR happening, at all. It might be forcing the game to render at the lower resolution, which would explain the performance gains, but it is not upscaling the results. So no, I'm afraid it is not working.

**Comment by walterklomp** [comment]
> I'm afraid I can't offer any further assistance. I don't own the game. But if there is no log file, that means that the game did not load vrperfkit's DLL. And I can think of no explanation that would make the game load Reshade's dxgi.dll, but not vrperfkit's.

Hi fholger, if I gift you ams 2 do you think you can get it to work? I need it desperately to work with hp reverb g2. The fov resize of steam makes it a cross-eyed mess so I think only the ffr might fix this?

**Comment by acollien** [comment]
> me is submitting D3D11 textures, creating necessary output resources..

Did this happen? Any progress made (*crosses fingers and toes*)

**Comment by stefan-girlich** [comment]
- facing the same issue with SteamVR; Win 10, Vive Pro 2, latest nVidia drivers
- [vrperfkit.log](https://github.com/fholger/vrperfkit/files/9668582/vrperfkit.log) appeared after using Ultimate-ASI-Loader [as suggested](https://github.com/fholger/vrperfkit/issues/16#issuecomment-1018376739)
- 100% performance improvement, but confirming that it is due to lower render resolution ([as stated before](https://github.com/fholger/vrperfkit/issues/16#issuecomment-1049368340))

it's quite a disappointment that this doesn't work; AMS2 in VR is one of the best gaming experiences of my life and i wish i could enjoy it for longer with a higher framerate.

i'd be happy to contribute to a digital game copy (or bug bounty, or beer, or all three!) for @fholger :)

**Comment by robertjamesdavies** [comment]
> * facing the same issue with SteamVR; Win 10, Vive Pro 2, latest nVidia drivers
> * [vrperfkit.log](https://github.com/fholger/vrperfkit/files/9668582/vrperfkit.log) appeared after using Ultimate-ASI-Loader [as suggested](https://github.com/fholger/vrperfkit/issues/16#issuecomment-1018376739)
> * 100% performance improvement, but confirming that it is due to lower render resolution ([as stated before](https://github.com/fholger/vrperfkit/issues/16#issuecomment-1049368340))
> 
> it's quite a disappointment that this doesn't work; AMS2 in VR is one of the best gaming experiences of my life and i wish i could enjoy it for longer with a higher framerate.
> 
> i'd be happy to contribute to a digital game copy (or bug bounty, or beer, or all three!) for @fholger :)

Guys, you can switch to OpenXR Toolkit now which has gone beyond this - Instructions on Reiza forum here - https://forum.reizastudios.com/threads/opencomposite-openxr.24915/
The latest version works well too - https://mbucchia.github.io/OpenXR-Toolkit/

**Comment by stefan-girlich** [comment]
thanks @robertjamesdavies , unfortunately OpenComposite does not support the vive pro 2 or any other devices based on steam vr ([source](https://docs.google.com/spreadsheets/d/1s2SSuRt0oHm91RUZB-R-ef5BrfOtvP_jwDwb6FuMF9Q/edit#gid=2068512515))

**Comment by dcg** [comment]
AMS2 is loading dinput8.dll. 
I am using a Vive Pro 2 and SteamVR.
What worked for me (beacuse I also use reshade):

- download https://github.com/ThirteenAG/Ultimate-ASI-Loader/releases/tag/v4.68   (x64) 
- put dinput8.dll in the same folder as the AMS2.exe
- rename reshade dll to reshade.asi
- rename vrperfkit dll (dxgi.dll) to vrperfkit.asi
- enjoy ;)


@stefan-girlich 
Cheers

**Comment by Toastbuster** [comment]
If anyone sees this please help. VRperfkit initialises the logs if AMS2.EXE is used but this does not launch the game in the QUEST 2 (only on monitor)
When AMS2AVX.EXE is used, the game launches in VR but there is no responce in the VRperf log file. 

Is there anyway I can get it to work for AMS2AVX.EXE?
